packages/spec-parser/src/manifestUpdater.ts

```diff
@@ -13,9 +13,193 @@ import {
+            const operationItem = (operations as any)[method] as OpenAPIV3.OperationObject;
+            if (operationItem) {
+              const operationId = operationItem.operationId!;
+              const description = operationItem.description ?? "";
+              const paramObject = operationItem.parameters as OpenAPIV3.ParameterObject[];
+              const requestBody = operationItem.requestBody as OpenAPIV3.ParameterObject;
+
+              const parameters: FunctionParameters = {
+                type: "object",
+                properties: {},
+                required: [],
+              };
+
+              if (paramObject) {
+                for (let i = 0; i < paramObject.length; i++) {
+                  const param = paramObject[i];
+
+                  const schema = param.schema as OpenAPIV3.SchemaObject;
+
+                  parameters.properties![param.name] = ManifestUpdater.mapOpenAPISchemaToFuncParam(
+                    schema,
+                    method,
+                    pathUrl
+                  );
+
+                  if (param.required) {
+                    parameters.required!.push(param.name);
+                  }
+
+                  if (!parameters.properties![param.name].description) {
+                    parameters.properties![param.name].description = param.description ?? "";
+                  }
+                }
+              }
+
+              if (requestBody) {
+                const requestJsonBody = requestBody.content!["application/json"];
+                const requestBodySchema = requestJsonBody.schema as OpenAPIV3.SchemaObject;
+
+                if (requestBodySchema.type === "object") {
+                  if (requestBodySchema.required) {
+                    parameters.required!.push(...requestBodySchema.required);
+                  }
+
+                  for (const property in requestBodySchema.properties) {
+                    const schema = requestBodySchema.properties[property] as OpenAPIV3.SchemaObject;
+                    parameters.properties![property] = ManifestUpdater.mapOpenAPISchemaToFuncParam(
+                      schema,
+                      method,
+                      pathUrl
+                    );
+                  }
+                } else {
+                  throw new SpecParserError(
+                    Utils.format(
+                      ConstantString.UnsupportedSchema,
+                      method,
+                      pathUrl,
+                      JSON.stringify(requestBodySchema)
+                    ),
+                    ErrorType.UpdateManifestFailed
+                  );
+                }
+              }
+
+              const funcObj: FunctionObject = {
+                name: operationId,
+                description: description,
+                parameters: parameters,
+                states: {
+                  reasoning: {
+                    description: ConstantString.ReasoningDescription,
```

Comment:
kind of curious why we use constant string here.  sample value: https://spec-hub.azurewebsites.net/specifications/PluginManifest-draft.html#example-7 Should we leave it empty if we are not able to generate it?

Reply:
Sure, we leave them empty